lib/ddtrace/configuration.rb

```diff
@@ -104,13 +98,28 @@ def reset!
     protected
 
     def components
-      COMPONENTS_LOCK.synchronize do
+      safely_synchronize do
         @components ||= build_components(configuration)
       end
     end
 
     private
 
+    def safely_synchronize
+      COMPONENTS_LOCK.synchronize do
+        begin
+          yield
+        rescue ThreadError => e
+          logger_without_components.warn(
```

Comment:
I think this is critical enough to be considered an error, given the tracer will be partially initialized when this happens.

Reply:
Updated in f6626c9608627cb6f1b04c8bf804426996830b4f